src/Dapr.PluggableComponents.Tests/Adaptors/PubSubAdaptorTests.cs

```diff
@@ -0,0 +1,343 @@
+            .Returns<PullMessagesResponse>(
+                async response =>
+                {
+                    Assert.Equal(contentType, response.ContentType);
+                    Assert.Equal(topic, response.TopicName);
+
+                    await reader.AddAsync(new PullMessagesRequest { AckError = new AckMessageError { Message = error }, AckMessageId = response.Id });
+                });
+
+        var pullMessagesTask = fixture.Adaptor.PullMessages(
+            reader,
+            mockWriter.Object,
+            fixture.Context);
+
+        await reader.AddAsync(new PullMessagesRequest { Topic = new Topic { Name = topic } });
+
+        await pullMessagesTask;
+
+        fixture.MockComponent
+            .Verify(
+                component => component.PullMessagesAsync(
+                    It.Is<PubSubPullMessagesTopic>(request => request.Name == topic),
+                    It.IsAny<MessageDeliveryHandler<string?, PubSubPullMessagesResponse>>(),
+                    It.Is<CancellationToken>(token => token == fixture.Context.CancellationToken)),
+                Times.Once());
+    }
+
+    [Fact(Timeout = TimeoutInMs)]
+    public async Task PullMessagesNoRequests()
+    {
+        using var fixture = AdaptorFixture.CreatePubSub();
+
+        fixture.MockComponent
+            .Setup(component => component.PullMessagesAsync(It.IsAny<PubSubPullMessagesTopic>(), It.IsAny<MessageDeliveryHandler<string?, PubSubPullMessagesResponse>>(), It.IsAny<CancellationToken>()));
+
+        var reader = new PullMessagesRequestStreamReader();
+        var mockWriter = new Mock<IServerStreamWriter<PullMessagesResponse>>();
+
+        var pullTask = fixture.Adaptor.PullMessages(
+            reader,
+            mockWriter.Object,
+            fixture.Context);
+
+        reader.Complete();
+
+        await Assert.ThrowsAsync<InvalidOperationException>(
+            async () =>
+            {
+                await pullTask;
+            }
+        );
+    }
+
+    [Fact(Timeout = TimeoutInMs)]
+    public async Task PullMessagesNoInitialTopic()
+    {
+        using var fixture = AdaptorFixture.CreatePubSub();
+
+        fixture.MockComponent
+            .Setup(component => component.PullMessagesAsync(It.IsAny<PubSubPullMessagesTopic>(), It.IsAny<MessageDeliveryHandler<string?, PubSubPullMessagesResponse>>(), It.IsAny<CancellationToken>()));
+
+        var reader = new PullMessagesRequestStreamReader();
+        var mockWriter = new Mock<IServerStreamWriter<PullMessagesResponse>>();
+
+        var pullTask = fixture.Adaptor.PullMessages(
+            reader,
+            mockWriter.Object,
+            fixture.Context);
+
+        await reader.AddAsync(new PullMessagesRequest { });
+
+        reader.Complete();
+
+        await Assert.ThrowsAsync<InvalidOperationException>(
+            async () =>
+            {
+                await pullTask;
+            }
+        );
```

Comment:
I prob missed it: why is `InvalidOperationException` thrown here?

Reply:
The exception should be thrown because a non-topic message was the first to be sent to the component.  (The adaptor expects a topic message following by n number of non-topic (i.e. ack) messages.)